sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -136,7 +133,7 @@ public Session createSession(@Nullable String defaultSchema) {
         private Throwable throwable = null;
         private List<List<Object>> bulkParams = new ArrayList<>();
         private List<ResultReceiver> resultReceivers = new ArrayList<>();
-        private Map<String, String> settings;
+        private Map<String, String> settings = new HashMap<>();
```

Comment:
Can you initialize this to an Collections.emptyMap and in the applySettings just change the assignment to the settings of the Set statements (maybe with a unmodifiableMap protection) ?